src/Scene.h

```diff
@@ -68,6 +73,12 @@ class CScene
 	bool intersect(Ray& ray) const
 	{
 		// --- PUT YOUR CODE HERE ---
+		for (auto prim_ : m_vpPrims) {
+			if (prim_->intersect(ray)) {
+				return true;
```

Comment:
here you find the _first_ intersection instead of finding _nearest_ intersection